ush/python/pygfs/task/marine_bmat.py

```diff
@@ -0,0 +1,364 @@
+
+    @logit(logger)
+    def initialize(self: Task) -> None:
+        """Initialize a global B-matrix
+
+        This method will initialize a global B-Matrix.
+        This includes:
+        - staging the deterministic backgrounds (middle of window)
+        - staging SOCA fix files
+        - staging static ensemble members (optional)
+        - staging ensemble members (optional)
+        - generating the YAML files for the JEDI and GDASApp executables
+        - creating output directories
+        """
+        super().initialize()
+
+        # stage fix files
+        logger.info(f"Staging SOCA fix files from {self.task_config.SOCA_INPUT_FIX_DIR}")
+        newdirs = [os.path.join(self.task_config.DATA, 'INPUT')]
+        FileHandler({'mkdir': newdirs}).sync()
+        soca_fix_list = parse_j2yaml(self.task_config.SOCA_FIX_YAML_TMPL, self.task_config)
+        FileHandler(soca_fix_list).sync()
+
+        # prepare the MOM6 input.nml
+        mdau.prep_input_nml(self.task_config)
+
+        # stage a single background
+        # TODO: check if we only need 1 deterministic background
+        # TODO: when we decide to move this task to the end of the previous cycle,
+        #       the bkg will have to come from HISTORY not HISTORY_PREV
+        ocn_bkg_dir = self.task_config.COMIN_OCEAN_HISTORY_PREV
+        ice_bkg_dir = self.task_config.COMIN_ICE_HISTORY_PREV
+        logger.info("Staging background files from {ocn_bkg_dir} and {ice_bkg_dir}")
+        bkg_list = []
+        bkg_list.append([os.path.join(ocn_bkg_dir, f"gdas.ocean.t{self.task_config.gcyc_str}z.inst.f009.nc"),
+                         os.path.join(self.task_config.DATA, "ocn.bkg.nc")])
+        bkg_list.append([os.path.join(ice_bkg_dir, f"gdas.ice.t{self.task_config.gcyc_str}z.inst.f009.nc"),
+                         os.path.join(self.task_config.DATA, "ice.bkg.nc")])
+        FileHandler({'copy': bkg_list}).sync()
+        mdau.cice_hist2fms("ice.bkg.nc", "ice.bkg.nc")
+
+        # Copy MOM6 restart
+        # TODO: check if we can combine this with the step above
+        logger.info(f"Linking MOM6 restart to ocn.bkg.nc")
+        rst_list = []
+        rst_list.append([os.path.join(self.task_config.DATA, "ocn.bkg.nc"),
+                         os.path.join(self.task_config.DATA, "INPUT", "MOM.res.nc")])
+        rst_list.append([os.path.join(self.task_config.DATA, "ice.bkg.nc"),
+                         os.path.join(self.task_config.DATA, "INPUT", "cice.res.nc")])
+        FileHandler({'copy': rst_list}).sync()
+
+        # stage the grid generation yaml
+        FileHandler({'copy': [[self.task_config.GRID_GEN_YAML,
+                               os.path.join(self.task_config.DATA, 'gridgen.yaml')]]}).sync()
+
+        # generate the variance partitioning YAML file
+        logger.debug("Generate variance partitioning YAML file")
+        diagb_config = parse_j2yaml(path=os.path.join(self.task_config.BERROR_YAML_DIR, 'soca_diagb.yaml.j2'),
```

Comment:
I think instead of defining BERROR_YAML_DIR and then hard-coded filenames, these should be user definable through the config.${job} file

Reply:
You're probably right, but for this PR I'd like to keep things hardcoded.